azure-quantum/azure/quantum/optimization/problem.py

```diff
@@ -147,6 +194,67 @@ def deserialize(
             problem.init_config = result["cost_function"]["initial_configuration"]
 
         return problem
+    
+    @classmethod
+    def deserialize_from_proto(
+        cls,
+        problem_as_str: Union[list,str],
+        name: Optional[str] = None
+    ) -> Problem:
+        msg_count = 0
+
+        problem = Problem(
+            name = name
+        )
+
+        for msg in problem_as_str:
+            proto_problem = problem_pb2.Problem()
+            proto_problem.ParseFromString(msg)
+            if msg_count == 0:
+                for qdk_type, proto_type in proto_types.items():
+                    if proto_problem.cost_function.type == proto_type:
+                        problem.problem_type = qdk_type
+                metadata = proto_problem.metadata               
+                if name is None:
+                    name = metadata["name"]
+                    problem.name = name
+            for msg_term in proto_problem.cost_function.terms:
+                term = Term(
+                    c = msg_term.c
+                )
+                term.ids = []
+                for msg_term_id in msg_term.ids:
+                    term.ids.append(msg_term_id)
+                problem.terms.append(term)
+        
+        return problem
+
+    @classmethod
+    def deserialize(
+        cls, 
+        problem_str: Union[list,str], 
+        name:Optional[str] = None, 
+        serialization_type:Optional[str] = None) -> Problem:
+        """Deserializes the problem from a
+        JSON string or protobuf messages serialized with Problem.serialize()
+        Also used to deserialize the messages downloaded from the blob
+
+        :param problem_str:
+            The string to be deserialized to a `Problem` instance
+        :type problem_str: str
```

Comment:
is the type for a problem serialized as protobuf really `str`? I was expecting this to be just Byte[]